Add exclude pattern cli arguments to in-toto-run/in-toto-record

lukpueh opened the pull request:
**Description of the changes being introduced by the pull request**:
in-toto provides an [`ARTIFACT_EXCLUDE_PATTERNS`](https://github.com/in-toto/in-toto/blob/0.2.1/README.md#settings) setting to exclude artifacts when recording link metadata.

This PR adds an `--exclude` argument to `in-toto-run` and `in-toto-record` to specify the patterns on the cli tool. Patterns passed via argument override patterns specified via setting (or envvar or RCfile).

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
Add exclude patterns arg to record artifacts

Add an optional keyword argument to record_artifacts_as_dict
to pass patterns used to exclude artifacts from being recorded.

So far, this was only available via the ARTIFACT_EXCLUDE_PATTERNS
setting.

The commit also updates and merges the related unittests.

Commit:
Add exclude patterns arg to runlib run and record

Update runlib's in_toto_run and in_toto_record_{start, stop}
functions to accept an optional exclude patterns argument and
pass it through to record_artifacts_as_dict, when creating link
metadata.

Commit:
Add exclude patterns arg to run and record cli

Add optional --exclude argument to in-toto-run and in-toto-record,
used to specify path patterns that should be excluded from the
resulting link metadata.
Exclude patterns passed via CLI take precedence over those
specified via ARTIFACT_EXCLUDE_PATTERNS setting.

SantiagoTorres commented on in_toto/in_toto_record.py:
```diff
@@ -56,6 +56,13 @@
                         default GPG keyring is used.
+  --exclude <pattern> [<pattern> ...]
+                        Do not record 'materials/products' matched by the
+                        passed exclude patterns. Passed exclude patterns
```
what do you think of
```
Do not record 'materials/products' that match <pattern>`
```
?

lukpueh replied:
Sounds good

SantiagoTorres commented on in_toto/in_toto_record.py:
```diff
@@ -172,6 +179,14 @@ def main():
+      "Do not record 'materials/products' matched by the passed exclude"
+      " patterns. Passed exclude patterns override previously set patterns,"
+      " using e.g.: environment variables or RCfiles. See"
+      " ARTIFACT_EXCLUDE_PATTERNS documentation for additional info."
```
I just realized that we need to repeat ourselves on these. I wonder if moving both strings to the same place would be worth it.

lukpueh replied:
100% agreed. We should create a ticket for generalizing our cli tools. There are a lot of shared arguments especially in `in-toto-run` and `in-toto-record`. And all tools take singing or verification key arguments for securesystemslib keys and gpg keys.

SantiagoTorres reviewed (commented):
Minor nits.

Commit:
Add module that defines common command line args

In this initial commit the module defines two constants, that
define the required *args and **kwargs to add an arpgarse
argument for the command line argument `--exclude` used in
in-toto-run and in-toto-record.

Commit:
Use common exclude argument in run/record cli

SantiagoTorres commented on in_toto/common_args.py:
```diff
@@ -0,0 +1,39 @@
+          " Passed exclude patterns override previously set patterns, using"
+          " e.g.: environment variables or RCfiles. See"
+          " ARTIFACT_EXCLUDE_PATTERNS documentation for additional info.")
+  }
```
sorry about the nitpick, can you add a newline?

lukpueh replied:
Done, thanks for the review @SantiagoTorres

SantiagoTorres reviewed (commented):
this minor change and LGTM
